spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/BackingServicesUpdateValidatorService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.workflow.instance;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.appbroker.deployer.BackingService;
+
+/**
+ * Validates updates to backing services, and returns an updated copy of the list with the previousPlan set. This help
+ * only triggering plan update when previousPlan is set.
+ */
+public class BackingServicesUpdateValidatorService {
+
+	// Accept and return Mono to make it easier for AppDeploymentUpdateServiceInstanceWorkflow to provide us with previous and candidate backing services
+	public Mono<List<BackingService>> validatePlanUpdate(Mono<List<BackingService>> previousBackingServices,
```

Comment:
`validatePlanUpdate` is not really just validating, actually, it doesn't really validate that the new parameters passed through the new `candidateBackingServicesForUpdate` are valid at all.

I would say this class is filtering the services to update, which is great, does one thing and does it well. Just the naming is what I find misleading.

Reply:
The actual validation is performed in the `getValidatedBackingService()` which will throw an exception, triggering the emission of an error in the flux.

Also, since the provided BackingService bean has not a request scope (i.e. is reused among requests), it returns a mutated copy with the updated plan set.

Alternative namings I can think of:
* `BackingServicesUpdateFactoryService`: expresses that a new `BackingService` list is created, but misses the fact that an exception is thrown when the user request is invalid/not supported
   * `validateAndMutatePlanUpdate()`could become `validateAndCreateUpdatedBackingServices()`

Please let me know if you find a better alternative naming and I'll then apply the change once (as to avoid back and forth changes)